plugins/twbt-utils.cpp

```diff
@@ -0,0 +1,122 @@
+#include "Core.h"
+#include "Console.h"
+#include "Export.h"
+#include "PluginManager.h"
+#include "VersionInfo.h"
+#include "VTableInterpose.h"
+#include "LuaTools.h"
+
+#include "DataDefs.h"
+
+#include "df/viewscreen_dwarfmodest.h"
+#include "df/init.h"
+#include "df/renderer.h"
+#include "df/graphic.h"
+#include "df/enabler.h"
+#include "df/map_renderer.h"
+
+using std::string;
+using std::vector;
+using namespace DFHack;
+using namespace df::enums;
+
+DFHACK_PLUGIN("twbt-utils");
+REQUIRE_GLOBAL(window_x)
+REQUIRE_GLOBAL(window_y)
+REQUIRE_GLOBAL(window_z)
+REQUIRE_GLOBAL_NO_USE(gps)
+REQUIRE_GLOBAL_NO_USE(enabler)
+REQUIRE_GLOBAL_NO_USE(twbt_render_map)
+
+#ifdef WIN32
+    // On Windows there's no parameter pointing to the map_renderer structure
+    typedef void(_stdcall *RENDER_MAP)(int);
+
+    RENDER_MAP _render_map;
+
+    void render_map(){ _render_map(0); }
+#else
+REQUIRE_GLOBAL_NO_USE(map_renderer)
+
+    typedef void(*RENDER_MAP)(void*, int);
+
+    RENDER_MAP _render_map;
+
+    void render_map(){ _render_map(df::global::map_renderer,0); }
+#endif
+static int render_map_rect(lua_State* L)
+{
+    CoreSuspender suspender;
+
+    int x = luaL_checkint(L, 1);
+    int y = luaL_checkint(L, 2);
+    int z = luaL_checkint(L, 3);
+    int w = luaL_checkint(L, 4);
+    int h = luaL_checkint(L, 5);
+    //backup state
+    uint8_t *s = df::global::gps->screen;
```

Comment:
Is this supposed to be a backup or just an alias?

Reply:
Initial idea was to have a std::vector for "local render buffer". However when i tried it, it did not work, not sure why though.
The rest of state (window_x/y/z, grid_x/y) is backuped.